### PR TITLE
Update emoji font size if it was updated in `markdownStyle`

### PR DESCRIPTION
**Pull request by VickyStash**
### Details

This PR fixes iOS issue, when emoji fontSize updates in the markdownStyle has no right-away effect on the emoji display.

### Related Issues

[GH_LINK](https://github.com/Expensify/react-native-live-markdown/issues/448)

### Manual Tests

1. Update [App.tsx](https://github.com/Expensify/react-native-live-markdown/blob/main/example/src/App.tsx) example with the next updates:
```
// Specify default markdownStyle value
const [markdownStyle, setMarkdownStyle] = React.useState({emoji: {fontSize: 19}});

// Display markdownStyles to see how it's updated
<Text>Markdown style: {JSON.stringify(markdownStyle)}</Text>

...

// Add two buttons to update emoji font size
<Button
  title="Update emoji font size to 27"
  onPress={() =>
    setMarkdownStyle({
      emoji: {
        fontSize: 27,
      },
    })
  }
/>
<Button
  title="Update emoji font size to 19"
  onPress={() =>
    setMarkdownStyle({
      emoji: {
        fontSize: 19,
      },
    })
  }
/>
```
2. Enter a couple of emojis and some text in the input
3. Tap on buttons to update the emoji font size. See that the emoji font size updated in the input right away.

https://github.com/user-attachments/assets/b0edf349-5baf-4a9b-9d48-86c28f5ed3e8

### Linked PRs

N/A

### CODE REVIEW

**Comment by VickyStash** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by VickyStash** [comment]
recheck

**Comment by tomekzaw** [comment]
Reviewing the PR

**Comment by tomekzaw** [comment]
@VickyStash Could you please merge main?

**Comment by tomekzaw** [comment]
@VickyStash Also, it looks like the change has been extracted from https://github.com/Expensify/react-native-live-markdown/pull/356, was that intentional? Do you think we should merge your PR first and then follow with https://github.com/Expensify/react-native-live-markdown/pull/356 or we should merge just https://github.com/Expensify/react-native-live-markdown/pull/356?

File: ios/MarkdownTextInputDecoratorView.mm [hunk 1]
```diff
@@ -110,12 +110,17 @@ - (void)setMarkdownStyle:(RCTMarkdownStyle *)markdownStyle
...
+      // We want to use `textStorage` for applying markdown when possible. Currently it's only available for UITextView
+      [_textView textDidChange];
+  } else {
   // apply new styles
 #ifdef RCT_NEW_ARCH_ENABLED
   [_textInput _setAttributedString:_backedTextInputView.attributedText];
 #else
   [_textInput setAttributedText:_textInput.attributedText];
 #endif /* RCT_NEW_ARCH_ENABLED */
+  }
```
Comment (tomekzaw) [inline ios/MarkdownTextInputDecoratorView.mm:123]: Let's add missing indentation
```suggestion
  if (_textView != nil) {
      // We want to use `textStorage` for applying markdown when possible. Currently it's only available for UITextView
      [_textView textDidChange];
  } else {
    // apply new styles
#ifdef RCT_NEW_ARCH_ENABLED
    [_textInput _setAttributedString:_backedTextInputView.attributedText];
#else
    [_textInput setAttributedText:_textInput.attributedText];
#endif /* RCT_NEW_ARCH_ENABLED */
  }
```

**Comment by VickyStash** [comment]
@tomekzaw I'm okay to merge the https://github.com/Expensify/react-native-live-markdown/pull/356 only and close this PR.
I've tried to confirm it the issue [comment](https://github.com/Expensify/react-native-live-markdown/issues/448#issuecomment-2338570881). I had some doubts why the https://github.com/Expensify/react-native-live-markdown/pull/356 was open from May.

File: ios/MarkdownTextInputDecoratorView.mm (hunk 1 above)
Comment (VickyStash) [reply ios/MarkdownTextInputDecoratorView.mm]: Done

**Comment by tomekzaw** [comment]
@VickyStash Thanks for your response. So it looks like your PR along with changes from https://github.com/Expensify/react-native-live-markdown/pull/474 indeed fixes the emoji font size issue on RN 0.75. However, there's still a problem when updating text font size:

https://github.com/user-attachments/assets/9e5f793c-6b39-42cf-8adf-11124753c1ad

If you're okay with closing this PR in favor of #356 then let's do that.

Thanks again for your investigation – https://github.com/Expensify/App/pull/47547#issuecomment-2345730522 is the best summary of what happened here.

**Comment by VickyStash** [comment]
Closed in favor of https://github.com/Expensify/react-native-live-markdown/pull/356

**Comment by tomekzaw** [comment]
Actually we've found one regression in https://github.com/Expensify/react-native-live-markdown/pull/356#issuecomment-2345808423 so we can't merge #356 in its current form.

Luckily, this regression doesn't happen on #475 (this PR):

https://github.com/user-attachments/assets/bdeabb3d-5793-4252-a67e-a505373e9bfd

So let's merge this PR as it fixes `emoji.fontSize` issue and does not break text suggestions.

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**